.github/workflows/prepare_release.yml

```
@@ -0,0 +1,65 @@
+---
+
+name: Prepare Release
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: 'Module version to be released.'
+        required: true
+        type: string
+      working-directory:
+        description: The working directory where all jobs should be executed. Used for modules in subdirectories like a monorepo or a control repository.
+        default: '.'
+        required: false
+        type: string
+      base-branch:
+        description: 'The branch that will be used as the origin for the release branch.'
+        required: false
+        default: 'master'
+        type: string
+
+env:
+  BUNDLE_WITHOUT: development:test:system_tests
+  GIT_AUTHOR_NAME: Release Automation
+  GIT_AUTHOR_EMAIL: "${{ github.repository_owner }}@users.noreply.github.com"
+
+jobs:
+  prepare_release:
+    defaults:
+      run:
+        working-directory: ${{ inputs.working-directory }}
+    name: 'Puppet Forge'
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.base-branch }}
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+          bundler-cache: true
+          working-directory: ${{ inputs.working-directory }}
+      - name: Update metadata.json to new version
+        run: bundle exec rake module:bump_to_version[${{ inputs.version }}]
```

https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#understanding-the-risk-of-script-injections recommends setting it to an environment variable instead of using shell interpolation. While the risk may be small here, I think it's still a good idea.

Perhaps we can even modify puppet-blacksmith to understand an environment variable. I'd be tempted to modify `module:bump` to support `BLACKSMITH_NEXT_VERSION` and otherwise simply bump a patch level. That would be around here: https://github.com/voxpupuli/puppet-blacksmith/blob/760c035c6b87861ca767358d96178f8ebdd6ec13/lib/puppet_blacksmith/rake_tasks.rb#L76-L88

And if you take that logic a step further, you can make the input version optional then. The workflow becomes:

* Call release prep without arguments
* Inspect the generated changelog
* Fix any labels
* Call the release workflow with the correct version number
* Merge PR
* Create tag

As a follow up: you could create a workflow that if the `release_prep` branch is merged that you create a tag automatically.

---

Turns out, there's already `module:bump:full` which uses `BLACKSMITH_FULL_VERSION`
```suggestion
        run: bundle exec rake module:bump:full
```

---

updated it